src/aspire/image/image.py

```diff
@@ -123,8 +122,10 @@ def res(self):
         return self.resolution
 
     def _check_key_dims(self, key):
-        if isinstance(key, tuple):
-            assert len(key) <= self._data.ndim
+        if isinstance(key, tuple) and (len(key) > self._data.ndim):
+            raise ValueError(
+                f"Image stack_dim is {self.stack_ndim}, slice length must be =< f{self.ndim}"
```

Comment:
This was my bad but the `f` in there isn't necessary.

Reply:
it was the `stack_ndim`, but I also missed the extra f :) I'll remove them thanks!